src/Sulu/Bundle/WebsiteBundle/Controller/SitemapController.php

```diff
@@ -86,9 +87,24 @@ public function indexAction(Request $request)
         if (!$response) {
             $sitemap = $this->xmlSitemapRenderer->renderIndex($request->getScheme(), $request->getHost());
             if (!$sitemap) {
-                $aliases = array_keys($this->sitemapProviderPool->getProviders());
+                $sitemapAlias = null;
 
-                return $this->sitemapPaginatedAction($request, reset($aliases), 1);
+                foreach ($this->sitemapProviderPool->getProviders() as $sitemapAlias => $provider) {
+                    if ($provider->getMaxPage($request->getScheme(), $request->getHost()) > 0) {
+                        $sitemapAlias = $provider->getAlias();
+
+                        break;
```

Comment:
This code is really confusing :thinking: I think that's because of the `SitemapProvider`... Are you checking in above's `if` if there is data at all? Does `getMaxPage` only return `0` if there is no data? 

Do we assume that if `renderIndex` above returns a falsy value, that no index has to be created and we return simply the first sitemap we find? Because if there would be two sitemaps we wouldn't get to this code anyway? :thinking:

Reply:
> Does getMaxPage only return 0 if there is no data?

Yes if a sitemap have no data for the current Domain it will return getMaxPage 0

> Do we assume that if renderIndex above returns a falsy value

yes this logic is strange `renderIndex` will return null if there is no need for an index file (only one provider does return has getMaxPage 0). Currently it did take with `reset()` the first provider but thats not always correct instead it should take the first provider which has data to provider thats why the loop was introduced.